src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -157,8 +203,15 @@ internal partial class Binder
         private TupleBinaryOperatorInfo.Multiple BindTupleBinaryOperatorNestedInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
             BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
         {
-            TypeSymbol leftType = left.Type;
-            TypeSymbol rightType = right.Type;
+            left = GiveTupleTypeToDefaultLiteralIfNeeded(left, right.Type);
+            right = GiveTupleTypeToDefaultLiteralIfNeeded(right, left.Type);
+
+            if ((left.Type is null && left.Kind != BoundKind.TupleLiteral) ||
+                (right.Type is null && right.Kind != BoundKind.TupleLiteral))
+            {
+                Error(diagnostics, ErrorCode.ERR_AmbigBinaryOps, node, node.OperatorToken.Text, left.Display, right.Display);
```

Comment:
>ERR_AmbigBinaryOps [](start = 45, length = 18)

It is not obvious why we consider the operator ambiguous in this case.

Reply:
The case here is really that we weren't able to give `default` a tuple type. 
But I wanted to write the condition a bit more broadly, so that the code below can clearly move ahead with assumption that we either have a tuple literal (which may be typeless) or a typed expression (on both sides).

I could add a comment or use a new error code if that feels more appropriate.

---
In reply to: [173291602](https://github.com/dotnet/roslyn/pull/25327#discussion_r173291602) [](ancestors = 173291602)

Reply:
> But I wanted to write the condition a bit more broadly

Consider if you can add an assert with more narrow condition that would verify your assumption.

---
In reply to: [173310337](https://github.com/dotnet/roslyn/pull/25327#discussion_r173310337) [](ancestors = 173310337,173291602)